### PR TITLE
OOP School Library Milestone #7: Unit tests

### PR DESCRIPTION
**Pull request by karlavdelgadof**
## OOP School Library Milestone #7: Unit tests :hammer_and_wrench: :school: :memo: :gear: 

These are the proposed changes :recycle:  for this project :arrow_down: :
- Created unit tests for the following classes as [specified in the requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md#:~:text=Add%20unit%20tests%20to%20your%20classes,while%20implementing%20the%20Decorator%20design%20pattern) :heavy_check_mark: :
   - Book :book: 
   - Person :person_curly_hair: 
   - Student :student: 
   - Teacher :teacher: 
   - Classroom :school: 
   - Rental :receipt: 
   - Decorator pattern classes (CapitalizeDecorator, TrimmerDecorator) :heart_decoration:

### CODE REVIEW

File: spec/student_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+describe Student do
+  before(:each) do
+    @student = Student.new(12, 'Agustina', false, '78')
+  end
+
+  it 'Should return the values provided to the constructor class' do
+    expect(@student.age).to eql 12
+    expect(@student.name).to eql 'Agustina'
+    expect(@student.parent_permission).to be false
+    expect(@student.classroom).to ql '78'
```
Comment (victorokoroji) [inline spec/student_spec.rb]: - [x] You have a failed test due to a misspelling. You spelled` eql` as` ql`, causing your test to fail. Kindly fix that 👍🏽 

image

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @karlavdelgadof, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/student_spec.rb (hunk 1 above)
Comment (karlavdelgadof) [reply spec/student_spec.rb]: Thank you @vickymarz is done ✅

**Review by Ismailco — APPROVED** [review]
### Hi @karlavdelgadof, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**